platform/api/aws/images.go

```diff
@@ -370,6 +370,43 @@ func (a *API) CreatePVImage(snapshotID string, diskSizeGiB uint, name string, de
 	return a.createImage(params)
 }
 
+// Remove all uploaded data associated with an AMI.
+func (a *API) RemoveImage(name, s3BucketName, s3ObjectPath string) error {
+	err := a.DeleteObject(s3BucketName, s3ObjectPath)
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok {
+			if awsErr.Code() != "NoSuchKey" {
+				return err
+			}
+		} else {
+			return err
+		}
+	}
+
+	imageID, err := a.FindImage(name)
+	if err != nil {
+		return err
+	}
+	if imageID != "" {
+		_, err := a.ec2.DeregisterImage(&ec2.DeregisterImageInput{ImageId: &imageID})
+		if err != nil {
+			return err
+		}
+	}
+
+	snapshot, err := a.FindSnapshot(name)
+	if err != nil {
+		return err
+	}
+	if snapshot != nil {
+		// We explicitly ignore errors here in case somehow another AMI was based
+		// on that snapshot
+		a.ec2.DeleteSnapshot(&ec2.DeleteSnapshotInput{SnapshotId: &snapshot.SnapshotID})
```

Comment:
Let's capture the error here and if it's not `nil` perform a `plog.Warnf("deleting snapshot %s: %v", snapshot.SnapshotID, err)`